docs/user-guide/command-line-interface.md

```diff
@@ -361,6 +362,16 @@ This option instructs ESLint to try to fix as many issues as possible. The fixes
 1. This option throws an error when code is piped to ESLint.
 1. This option has no effect on code that uses processors.
 
+If you want to fix code from stdin or otherwise want to get the fixes without actually writing them to the file, use the [`--fix-dry-run`](--fix-dry-run) option.
+
+#### `--fix-dry-run`
+
+This option has the same effect as `--fix` with one difference: the fixes are not being saved to the file system. This makes fixing the code from stdin possible.
+
+Because the default formatter does not output the fixed code, you'll have to use another one (e.g. `json`) to get the fixes.
+
+This flag was added with editor plugins in mind - in some cases they can only use stdio for linting. Without this option enabling autofixing in such tools is close to impossible.
```

Comment:
I think this paragraph could be rephrased a bit. The main goal of the paragraph is to provide an example use case for the flag, so it doesn't seem useful to describe what editor plugins would be like if this flag didn't exist.

For example:

> This flag can be useful for integrations (e.g. editor plugins) which need to autofix text from the command line without saving it to the filesystem.